ignite/handlers/checkpoint.py

```diff
@@ -15,14 +15,39 @@
 
 
 class BaseSaveHandler(metaclass=ABCMeta):
-    """Base class for save handlers"""
+    """Base class for save handlers
+
+    Methods to override:
+
+    - :meth:`~ignite.handlers.checkpoint.BaseSaveHandler.__call__`
+    - :meth:`~ignite.handlers.checkpoint.BaseSaveHandler.remove`
+    """
 
     @abstractmethod
-    def __call__(self, checkpoint: Mapping, filename: str) -> None:
+    def __call__(self, checkpoint: Mapping, filename: str, metadata: Mapping = None) -> None:
```

Comment:
Should we reflect `metadata` as optional with `typing.Optional`?

Reply:
Yes, correct! Thanks @erip